<?xml version="1.0" encoding="UTF-8"?>
<!-- 开发期测试专用流程定文件 -->
<process id="Proc_Develop_Test" name="TestXmlforDeveloping">
	<!-- 流程开始 -->
	<start name="start">
		<transition to="task01"/>
	</start>
	
	<!-- step:第一个任务 -->
	<task name="task01" refdubbo="OspDubboService"
		refcomponent="TaskService01">
		<!-- field标签内的内容将会以k-v对的形式给到task，当做task的入口参数 -->
		<field key="serviceID" value="serviceID-20160729-719931" />
		<transition to="fork01"/>
	</task>
	
	<fork name="fork01">
		<transition to="task02" />
		<transition to="task03" />
	</fork>
	
	<!-- step:第二个测试任务 -->
	<task name="task02" refdubbo="OspDubboService"
		refcomponent="TaskService02">
		<transition to="join01"/>
	</task>
	
	<!-- step:第三个测试任务 -->
	<task name="task03" refdubbo="OspDubboService"
		refcomponent="TaskService03">
		<transition to="join01"/>
	</task>
		
	<join name="join01">
		<transition to="task04" />
	</join>
	
	<!-- step:第四个测试任务 -->
	<task name="task04" refdubbo="OspDubboService"
		refcomponent="TaskService04">
		<transition to="end"/><!-- 正常流程就直接结束 -->
		<!-- 若抛出IOException类型的异常，则跳转到task05 -->
		<!-- 该异常会流传至后面的task -->
		<!-- 每次执行过task会记录调用次数，到达上限之后，异常分支将不再执行，直接抛出，该task失败 -->
		<exception  type="BusinessException" to="task05"/>
		<!-- 若抛出BusinessException类型的异常，则跳转到task06 -->
		<exception type="IOException"  to="task06"/>
	</task>
		
	<!-- step:第五个测试任务 -->
	<task name="task05" refdubbo="OspDubboService"
		refcomponent="TaskService05">
		<transition to="task01"/>
	</task>
	
	<!-- step:第六个测试任务 -->
	<task name="task06" refdubbo="OspDubboService"
		refcomponent="TaskService06">
		<transition to="end"/>
	</task>	
	
	<!-- 流程结束 -->
	<end name="end"/>
</process>